src/main/java/juja/microservices/gamification/service/AchievementService.java

```diff
@@ -58,55 +57,76 @@ public String addDaily(DailyRequest request) {
             logger.debug("Added new description to old description");
             achievement.setDescription(description);
             logger.debug("Description is set to achievement {}", achievement.getId());
-            logger.info("Added description to daily achivement from user '{}'", userFromId);
 
+            logger.info("Added description to daily achivement from user '{}'", userFromId);
             return achievementRepository.addAchievement(achievement);
         }
     }
 
     public List<String> addThanks(ThanksRequest request) {
-        requestValidator.checkThanks(request);
-
         String userFromId = request.getFrom();
         String userToId = request.getTo();
         String description = request.getDescription();
+
+        if (userFromId.equalsIgnoreCase(userToId)) {
+            logger.warn("User '{}' trying to put 'Thanks' achievement to yourself", request.getTo());
+            throw new UnsupportedAchievementException("You cannot thank yourself");
+        }
+
         List<Achievement> userFromAndToListToday = achievementRepository
                 .getAllAchievementsByUserFromIdCurrentDateType(userFromId, AchievementType.THANKS);
 
         for (Achievement achievement : userFromAndToListToday) {
-            if (achievement.getTo().equals(userToId)) {
-                logger.warn("User '{}' tried to give 'Thanks' achievement more than one times to person '{}'", userFromId, userToId);
-                throw new UnsupportedAchievementException("You cannot give more than one thanks for day one person");
-            }
+            checkTryToThanksYourself(userFromId, userToId, achievement);
         }
 
-        List<String> result = new ArrayList<>();
         if (userFromAndToListToday.isEmpty()) {
-            Achievement firstThanks = new Achievement(userFromId, userToId, 1, description, AchievementType.THANKS);
-            result.add(achievementRepository.addAchievement(firstThanks));
-            logger.info("Added first 'Thanks' achievement from user '{}' to user '{}'", userFromId, userToId);
-            return result;
+            return addFirstThanks(userFromId, userToId, description);
```

Comment:
It can be better if the order of 'if' statements will be: 1) addFirst, 2) addSecond, 3) Exception. Or 1) Exception, 2)addFirst, 3)addSecond.